### PR TITLE
feat: hook out-of-band svc into didexchange svc

### LINKED ISSUE #1491: outofband protocol: hook into did-exchange (temporary)
part of #1454 

Hook the OOB protocol service into the didexchange service by having the former pass in a `didexchange.Invitation` object to the latter.

This is a temporary solution to minimize refactors. In the longer term, the didexchange protocol service should be refactored to accept an invitation object with a different shape than `didexchange.Invitation`'s.

### PR DESCRIPTION
**Pull request by llorllale**
closes #1491 

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

File: pkg/didcomm/protocol/outofband/service_test.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
```
Comment (troyronda) [inline pkg/didcomm/protocol/outofband/service_test.go]: should be in the third import group.

File: pkg/didcomm/protocol/outofband/service.go [hunk 2]
```diff
@@ -209,23 +211,23 @@ func listener(
 }
 
 func (s *Service) handleRequestCallback(c *callback) error {
-	// TODO either transform to a didexchange.invitation object or refactor didexchange.Service to accept this object
-	connID, err := s.didSvc.HandleInbound(c.msg, c.myDID, c.theirDID)
+	// TODO refactor didexchange.Service to accept an object other than didexchange.Invitation
```
Comment (troyronda) [inline pkg/didcomm/protocol/outofband/service.go]: issue number?

Comment (llorllale) [reply pkg/didcomm/protocol/outofband/service.go]: #1501

File: pkg/didcomm/protocol/outofband/service_test.go (hunk 1 above)
Comment (llorllale) [reply pkg/didcomm/protocol/outofband/service_test.go]: done

File: pkg/didcomm/protocol/outofband/service.go [hunk 3]
```diff
@@ -335,3 +338,33 @@ func extractDIDCommMsgBytes(_ *decorator.Attachment) ([]byte, error) {
...
+			invitation.DID = did
+			break
+		}
+	}
+
+	if invitation.DID == "" {
+		return nil, nil, errors.New("ouf-of-band protocol only supports invitations with public DIDs for now")
+	}
+
+	// TODO support explicit invitations
```
Comment (troyronda) [inline pkg/didcomm/protocol/outofband/service.go]: issue number?

File: pkg/didcomm/protocol/outofband/service_test.go [hunk 4]
```diff
@@ -585,3 +664,39 @@ func newAck(pthid ...string) *model.Ack {
...
 	return a
 }
+
+type testDIDCommMsg struct {
+	errDecode error
+}
+
+func (t *testDIDCommMsg) ID() string {
+	panic("implement me")
+}
```
Comment (soluchok) [inline pkg/didcomm/protocol/outofband/service_test.go]: I would recommend you to replace it with `gomock`

Comment (llorllale) [reply pkg/didcomm/protocol/outofband/service_test.go]: noted - maybe in the future

**Pull request merged**